xs/src/libslic3r/PerimeterGenerator.cpp

```diff
@@ -85,47 +85,81 @@ PerimeterGenerator::process()
             for (int i = 0; i <= loop_number+1; ++i) {  // outer loop is 0
                 Polygons offsets;
                 if (i == 0) {
-                    // the minimum thickness of a single loop is:
-                    // ext_width/2 + ext_spacing/2 + spacing/2 + width/2
+                    // compute next onion, without taking care of thin_walls : destroy too thin areas.
+                    if (!this->config->thin_walls)
+                        offsets = offset(last, -(float)(ext_pwidth / 2));
+
+
+                    // look for thin walls
                     if (this->config->thin_walls) {
+                        // the minimum thickness of a single loop is:
+                        // ext_width/2 + ext_spacing/2 + spacing/2 + width/2
                         offsets = offset2(
                             last,
                             -(ext_pwidth/2 + ext_min_spacing/2 - 1),
-                            +(ext_min_spacing/2 - 1)
-                        );
-                    } else {
-                        offsets = offset(last, -ext_pwidth/2);
-                    }
-                    
-                    // look for thin walls
-                    if (this->config->thin_walls) {
-                        Polygons no_thin_zone = offset(offsets, +ext_pwidth/2);
-                        Polygons diffpp = diff(
-                            last,
-                            no_thin_zone,
-                            true  // medial axis requires non-overlapping geometry
-                        );
+                            +(ext_min_spacing/2 - 1));
+
+                        // detect edge case where a curve can be split in multiple small chunks.
+                        Polygons no_thin_onion = offset(last, -(float)(ext_pwidth / 2));
+                        float div = 2;
```

Comment:
div == 2 is not used, it is increased first.

Reply:
yes.